### PR TITLE
Added Content Mapping and Alpha Rewarded Interstitial Ad

### PR DESCRIPTION
**Pull request by aishikappyhigh**
Added Content Mapping for native and banner Ads
Updated All the Models
[ALPHA] Added Rewarded Interstitial Ad
Fixes - Fixed A Native Layout Design

### CODE REVIEW

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt [hunk 1]
```diff
@@ -1177,21 +1218,26 @@ object AdSdk {
...
+        loadAd(adLoader, context, adUnit, "e", contentURL, neighbourContentURL)
     }
 
 
     private fun loadAd(
         adLoader: AdLoader?,
         context: Context,
         adUnit: String,
-        s: String
+        s: String,
```
Comment (rajatkumardhamija) [inline adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: What is this s variable?

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt [hunk 2]
```diff
@@ -1214,7 +1260,7 @@ object AdSdk {
                     val iconHeight = mediaMaxHeight
                     iconView1.layoutParams = LinearLayout.LayoutParams(1, iconHeight)
                 }
-                iconView1.visibility = View.GONE
+//                iconView1.visibility = View.GONE
```
Comment (rajatkumardhamija) [inline adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: remove comments

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt [hunk 3]
```diff
@@ -1312,6 +1358,76 @@ object AdSdk {
...
+                }
+            }
+        }
+        ctd.start()
+        val builder = AlertDialog.Builder(activity, R.style.DialogTheme)
+        builder.setView(
+            LayoutInflater.from(activity).inflate(R.layout.ad_loading_layout_inters, null)
+        )
+        create = builder.create()
+        create?.show()
```
Comment (rajatkumardhamija) [inline adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: add is finishing check

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt [hunk 4]
```diff
@@ -1312,6 +1358,76 @@ object AdSdk {
...
+        RewardedInterstitialAd.load(activity, adId,
+            AdRequest.Builder().build(), object : RewardedInterstitialAdLoadCallback() {
+                override fun onAdLoaded(ad: RewardedInterstitialAd) {
+                    ctd.cancel()
+                    if (!moved) {
+                        Log.d(TAG, "Ad was loaded.")
+                        create?.dismiss()
```
Comment (rajatkumardhamija) [inline adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: ad is showing check and is finishing check

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt [hunk 5]
```diff
@@ -1312,6 +1358,76 @@ object AdSdk {
...
+                                    interstitialCallback.moveNext()
+                                    moved = true
+                                }
+                            }
+                        }
+                    }
+                }
+
+                override fun onAdFailedToLoad(adError: LoadAdError) {
+                    create?.dismiss()
```
Comment (rajatkumardhamija) [inline adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: add is showing and is finishing check

**Review by rajatkumardhamija — CHANGES_REQUESTED** [review]
Review

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt (hunk 1 above)
Comment (aishikappyhigh) [reply adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: removed them as it was mainly used for debug logging to identify the ad units and location

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt (hunk 2 above)
Comment (aishikappyhigh) [reply adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: done

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt (hunk 3 above)
Comment (aishikappyhigh) [reply adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: done

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt (hunk 4 above)
Comment (aishikappyhigh) [reply adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: done

File: adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt (hunk 5 above)
Comment (aishikappyhigh) [reply adutils/src/main/java/com/appyhigh/adutils/AdSdk.kt]: done

**Pull request closed without merging**